code/game/turfs/snow.dm

```diff
@@ -168,19 +163,22 @@
 
 	switch(burnlevel)
 		if(1 to 10)
-			slayer--
+			slayer = max(0, slayer - 1)
+			update_icon(1, 0)
 		if(11 to 24)
 			slayer = max(0, slayer - 2)
+			update_icon(1, 0)
 		if(25 to INFINITY)
 			slayer = 0
+			update_icon(1, 0)
```

Comment:
why repeat the update_icon everytime ? just put it out of the chain

Reply:
Copy pasta tbh but yeah

Reply:
if possible do on the other one that you copypasted from